vividus-plugin-azure-resource-manager/src/main/java/org/vividus/azure/configuration/AzureKeyVaultPropertiesProcessor.java

```diff
@@ -0,0 +1,86 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.azure.configuration;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.profile.AzureProfile;
+
+import org.apache.commons.lang3.Validate;
+import org.vividus.azure.client.AzureResourceManagerClient;
+import org.vividus.azure.identity.CredentialFactory;
+import org.vividus.configuration.AbstractPropertiesProcessor;
+import org.vividus.util.json.JsonPathUtils;
+
+public class AzureKeyVaultPropertiesProcessor extends AbstractPropertiesProcessor
+{
+    private static final String ENVIRONMENT_PROPERTY = "azure.environment";
+    private static final String KEY_VAULT_NAME_PROPERTY = "azure.key-vault.name";
+    private static final String KEY_VAULT_API_VERSION_PROPERTY = "azure.key-vault.api-version";
+
+    private static final Map<String, AzureEnvironment> ENVIRONMENT_MAP = Map.of(
+            "AZURE", AzureEnvironment.AZURE,
+            "AZURE_CHINA", AzureEnvironment.AZURE_CHINA,
+            "AZURE_GERMANY", AzureEnvironment.AZURE_GERMANY,
+            "AZURE_US_GOVERNMENT", AzureEnvironment.AZURE_US_GOVERNMENT
+    );
+
+    private AzureEnvironment azureEnvironment;
+    private String keyVaultName;
+    private String apiVersion;
+    private AzureResourceManagerClient client;
+
+    public AzureKeyVaultPropertiesProcessor()
+    {
+        super("AZURE_KEY_VAULT");
+    }
+
+    @Override
+    public Properties processProperties(Properties properties)
+    {
+        this.azureEnvironment = ENVIRONMENT_MAP.get(properties.getProperty(ENVIRONMENT_PROPERTY));
+        this.keyVaultName = properties.getProperty(KEY_VAULT_NAME_PROPERTY);
+        this.apiVersion = properties.getProperty(KEY_VAULT_API_VERSION_PROPERTY);
+        return super.processProperties(properties);
+    }
+
+    @Override
+    protected String processValue(String propertyName, String partOfPropertyValueToProcess)
+    {
+        return getKeyVaultSecretValue(partOfPropertyValueToProcess);
+    }
+
+    private String getKeyVaultSecretValue(String secretName)
+    {
+        if (client == null)
+        {
+            Validate.notEmpty(keyVaultName, "Provide value into property `%s`", KEY_VAULT_NAME_PROPERTY);
+            this.client = new AzureResourceManagerClient(new AzureProfile(azureEnvironment),
+                    CredentialFactory.createTokenCredential(), null);
+        }
+        String[] value = new String[1];
+        client.executeHttpRequest(HttpMethod.GET,
+                String.format("https://%s%s/secrets/%s?api-version=%s", keyVaultName,
+                        azureEnvironment.getKeyVaultDnsSuffix(), secretName, apiVersion), Optional.empty(),
+                response -> value[0] = JsonPathUtils.getData(response, "$.value"));
```

Comment:
client should be updated to use not `Consumer`, but `Function` for response body handling